## Haplotype data
For each species, the respective bam file, gene reference file, and associated index file are kept in the directory with the species name. 
The sorted bam files were visualized in igv and checked for a polymorphic site between two haplotypes. 
The read groups were added based on the changes in the polymorphic sites, which is provided in the tsv input file. Read group added in bam file using a jvarkit. Their respective plots were generated using the script.sh file.
