## Haplotype data
For each species, the respective bam file, gene reference file and it's associated index file are kept in the directory with species name. 
The sorted bam files were visualized in igv and checked for polymorphic site between two haplotypes. 
The read groups were added based on the changes in the polymorphic sites which is provided in tsv input file. 
For read group addition bam files we used jvarkit.
Their respective plot were generated using the script.sh file kept in this directory.
